cos_registration_agent/machine_ip_address.py

```diff
@@ -14,6 +14,16 @@ def get_machine_ip_address(url: str) -> str:
         parsed_url = urllib.parse.urlparse(url)
         host = parsed_url.hostname
 
+        if host is None:
+            raise ValueError(f"Invalid URL, no hostname found: {url}")
+
+        # Resolve hostname to IP address if needed
+        try:
+            host = socket.gethostbyname(host)
+        except socket.gaierror as e:
+            logger.error(f"Failed to resolve hostname {host}: {e}")
+            raise
```

Comment:
Should we really re-raise the exception from `socket`? This would mean import the `socket` lib in any file using this.
I would rather see a raise of a new exception like a `SystemError`.
What do you think?

Reply:
Agree, and changed in ac63ae72b54f147cacc161cffbd18ee76772d7de